src/Core__String.resi

```diff
@@ -0,0 +1,735 @@
+on MDN.
+```rescript
+String.charAt("Reason", 0) == "R"
+String.charAt("Reason", 12) == ""
+String.charAt(`Rẽasöń`, 5) == `ń`
+```
+*/
+@send
+external charAt: (string, int) => string = "charAt"
+
+/**
+`charCodeAt(str, n)` returns the character code at position `n` in string `str`
+the result is in the range 0-65535, unlke `codePointAt`, so it will not work
+correctly for characters with code points greater than or equal to 0x10000. The
+return type is `float` because this function returns NaN if `n` is less than
+zero or greater than the length of the string.
+See [`String.charCodeAt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt)
+on MDN.
+```rescript
+String.charCodeAt(`😺`, 0) == 0xd83d->Belt.Int.toFloat
+String.codePointAt(`😺`, 0) == Some(0x1f63a)
+```
+*/
+@send
+external charCodeAt: (string, int) => float = "charCodeAt"
+
+/**
+`codePointAt(str, n)` returns the code point at position `n` within string `str`
+as a `Some(value)`. The return value handles code points greater than or equal
+to 0x10000. If there is no code point at the given position, the function
+returns `None`.
+See [`String.codePointAt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt)
+on MDN.
+```rescript
+Js.String2.codePointAt(`¿😺?`, 1) == Some(0x1f63a)
+Js.String2.codePointAt("abc", 5) == None
+```
+*/
+@send
+external codePointAt: (string, int) => option<int> = "codePointAt"
+
+/**
+`concat(original, append)` returns a new `string` with `append` added after
+`original`.
+See [`String.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat)
+on MDN.
+```rescript
+String.concat("cow", "bell") == "cowbell"
+```
+*/
+@send
+external concat: (string, string) => string = "concat"
+
+/**
+`concatMany(original, arr)` returns a new `string` consisting of each item of an
+array of strings added to the `original` string.
+See [`String.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat)
+on MDN.
+```rescript
+String.concatMany("1st", ["2nd", "3rd", "4th"]) == "1st2nd3rd4th"
+```
+*/
+@variadic
+@send
+external concatMany: (string, array<string>) => string = "concat"
+
+/**
+ES2015: `endsWith(str, substr)` returns `true` if the `str` ends with `substr`,
+`false` otherwise.
+See [`String.endsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)
+on MDN.
+```rescript
+String.endsWith("BuckleScript", "Script") == true
+String.endsWith("BuckleShoes", "Script") == false
+```
+*/
+@send
+external endsWith: (string, string) => bool = "endsWith"
+
+/**
+`endsWithFrom(str, ending, len)` returns `true` if the first len characters of
+`str` end with `ending`, `false` otherwise. If `len` is greater than or equal
+to the length of `str`, then it works like `endsWith`. (Honestly, this should
+have been named endsWithAt, but oh well).
```

Comment:
Should be renamed to `endsWithAt`?

Reply:
No let's no change it now. But remove that part of the comment perhaps? Doesn't feel like a comment like that belongs in the docstring.